controller/client/client.go

```diff
@@ -492,7 +492,7 @@ func (c *Client) StreamDeployment(d *ct.Deployment, output chan *ct.DeploymentEv
 	}, appEvents)
 }
 
-func (c *Client) DeployAppRelease(appID, releaseID string) error {
+func (c *Client) DeployAppRelease(appID, releaseID string, logf func(*ct.DeploymentEvent)) error {
```

Comment:
I'm not sold on the function argument, perhaps this function should return an optional `stream.Stream` if a channel is provided? /cc @lmars